src/modules/user/UserContext.jsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, useState, useContext, useEffect } from "react";
+
+export const UserContext = createContext();
+export const UserDispatchContext = createContext();
+
+export const useUser = () => useContext(UserContext);
+
+export const UserProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
```

Comment:
Using state is fine for small projects. When you work on a bigger project, using reducer pattern (useReducer hook as we did in the lessons) would be more appropriate.

It's very good that you defined different functions (login, logout) direct in the UserProvider component and share them outside the component as ready to use functions. 👏

Reply:
I have noted down all of them for the future. Thanks =)